File: python/leetcode/add_binary.py
class Solution(object):
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        length = max(len(a), len(b))
        compl = 0
        res = ""
        for i in range(1, length + 1):
            sum_a = 0 if i > len(a) else int(a[-i])
            sum_b = 0 if i > len(b) else int(b[-i])
            _sum = sum_a + sum_b + compl
            compl = 1 if _sum > 1 else 0
            res += str(_sum % 2)
        if compl: res += str(compl)
        return res[::-1]

File: src/main/java/LeetCode/AddBinary.java
package LeetCode;

public class AddBinary {
    public String addBinary(String a, String b) {
        int i = a.length() - 1;
        int j = b.length() - 1;
        StringBuilder res = new StringBuilder();
        int compl = 0;
        while (i >= 0 || j >= 0) {
            int sum_a = i < 0 ? 0 : a.charAt(i--) - '0';
            int sum_b = j < 0 ? 0 : b.charAt(j--) - '0';
            int _sum = sum_a + sum_b + compl;
            compl = _sum > 1 ? 1 : 0;
            res.append(_sum % 2);
        }
        if (compl != 0) res.append(compl);
        return res.reverse().toString();
    }
}
